File: 프로그래머스/2/176962. 과제 진행하기/과제 진행하기.js
function solution(plans) {
    let answer = [];
    
    plans.sort((a, b) => convertM(a) - convertM(b));
    
    for(const plan of plans)
        console.log(plan[0]);
    
    function convertM(plan) {
        const time = plan[1].split(':');
        
        return parseInt(time[0]) * 60 + parseInt(time[1]);
    }
    
    let tmp = [];
    let time = 0;
    for(const plan of plans) {
        let start = convertM(plan);
        while (time < start) {
            if(tmp.length !== 0){
                tmp[tmp.length - 1][2]--;
                
                if(tmp[tmp.length - 1][2] === 0) {
                    answer.push(tmp[tmp.length - 1][0]);
                    tmp.pop();
                }
            }

            time++;
        }
        
        tmp.push(plan);
    }

    while(tmp.length !== 0) {
        answer.push(tmp[tmp.length - 1][0]);
        tmp.pop();
    }
    
    return answer;
}

File: 프로그래머스/2/176962. 과제 진행하기/과제 진행하기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <sstream>
#include <iostream>

using namespace std;

bool Comp(vector<string> a, vector<string> b) {
    return a[1] < b[1];
}

int startTime(vector<string> plan) {
    istringstream ss(plan[1]);
    string tmp;
    vector<int> v;

    while(getline(ss, tmp, ':')) {
        v.push_back(stoi(tmp));
    }
    
    return v[0] * 60 + v[1];
}

vector<string> solution(vector<vector<string>> plans) {
    vector<string> answer;
    vector<pair<string, int>> tmp;
    
    sort(plans.begin(), plans.end(), Comp);
    
    for(auto plan: plans)
        cout<<plan[0]<<' ';
    
    int time = 0;
    for(auto plan: plans) {
        int start = startTime(plan);
        
        while(time < start) {
            if(!tmp.empty()) {
                tmp.back().second--;
                if(tmp.back().second == 0){
                    answer.push_back(tmp.back().first);
                    tmp.pop_back();
                }
            }
            
            time++;
        }
        
        tmp.push_back({ plan[0], stoi(plan[2]) });
    }
    
    while(!tmp.empty()) {
        answer.push_back(tmp.back().first);
        tmp.pop_back();
    }
    
    return answer;
}